cpp/tests/c_api/random_walks_test.c

```diff
@@ -0,0 +1,434 @@
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "weight device_buffer update failed.");
+
+  ret_code = cugraph_make_device_buffer(p_handle, vertex_tid, num_paths, &dbuf_start);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "start device_buffer creation failed.");
+
+  ret_code = cugraph_update_device_buffer(p_handle, vertex_tid, &dbuf_start, (byte_t*)h_start);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "start device_buffer update failed.");
+
+  p_sampling_strategy = cugraph_create_sampling_strategy(0, 0.0, 0.0);
+  runtime_assert(p_sampling_strategy != NULL, "sampling strategy creation failed.");
+
+  p_graph_envelope = cugraph_make_sg_graph(p_handle,
+                                           vertex_tid,
+                                           edge_tid,
+                                           weight_tid,
+                                           FALSE,
+                                           &dbuf_src,
+                                           &dbuf_dst,
+                                           &dbuf_wgt,
+                                           num_vertices,
+                                           num_edges,
+                                           FALSE,
+                                           FALSE,
+                                           FALSE);
+  runtime_assert(p_graph_envelope != NULL, "graph envelope creation failed.");
+
+  ret_code = cugraph_random_walks(p_handle,
+                                  p_graph_envelope,
+                                  &dbuf_start,
+                                  num_paths,
+                                  max_depth,
+                                  flag_use_padding,
+                                  p_sampling_strategy,
+                                  &rw_ret);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "cugraph_random_walks() failed.");
+
+  extract_size_rw_result(&rw_ret, &dbuf_rw_ret_sz);
+
+  ret_code = cugraph_update_host_buffer(p_handle, edge_tid, (byte_t*)h_sizes, &dbuf_rw_ret_sz);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "size host buffer update failed.");
+
+  extract_vertex_rw_result(&rw_ret, &dbuf_rw_ret_v);
+
+  ret_code = cugraph_update_host_buffer(p_handle, vertex_tid, (byte_t*)h_paths, &dbuf_rw_ret_v);
+  runtime_assert(ret_code == CUGRAPH_SUCCESS, "paths host buffer update failed.");
+
+  flag_passed = host_check_paths(h_paths, h_sizes, num_paths);
+  runtime_assert(flag_passed == TRUE, "paths check failed.");
+
+  cugraph_free_rw_result(&rw_ret);
+
+  cugraph_free_graph(p_graph_envelope);
+
+  cugraph_free_sampling_strategy(p_sampling_strategy);
+
+  cugraph_free_device_buffer(&dbuf_start);
+
+  cugraph_free_device_buffer(&dbuf_wgt);
+
+  cugraph_free_device_buffer(&dbuf_dst);
+
+  cugraph_free_device_buffer(&dbuf_src);
+
+  cugraph_free_handle(p_handle);
+
+  return 0;
+}
+
+/* negative test RW call flow*/
+int test_random_walks_3()
```

Comment:
this tests and others still using the placeholder name need to have a more descriptive name.

Reply:
You provided these names. You're free to change them in the next release.